### ISSUE TITLE
The accessibility test is failing on Datagrid component

### ISSUE THREAD

**Issue by seonyukim** [carbon-design-system/ibm-products#3184]
## What package(s) are you using?

- [X] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

> Describe in detail the issue you're having.
The DataGrid has an accessibility test violation. When the table is rendered it doesn't take aria-label  for `<table>` and `<th>` .

> Is this issue related to a specific component?
DataGrid
> What did you expect to happen? What happened instead? What would you like to
> see changed?
No accessibility test failure for Datagrid
> What browser are you working in?
Chrome
> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?
"2.1.2"
> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?
As soon as possible
## Steps to reproduce the issue

1. Step one
2. Step two
3. Step three
4. etc.

> Please create a reduced test case in CodeSandbox
>
> https://codesandbox.io/s/github/carbon-design-system/ibm-products/tree/main/examples/carbon-for-ibm-products/SidePanel

## Additional information
![Screenshot 2023-07-07 at 2 28 01 PM](https://github.com/carbon-design-system/ibm-products/assets/14826482/f1bdf30b-e842-4cd4-886a-c0ab5a1c31a7)

- Screenshots or code
- Notes

**Comment by seonyukim** [comment]
Do you mean this is fixed v2? I initially found from V2

**Comment by davidmenendez** [comment]
yes it's fixed in v2. it should be in our latest release

**Comment by seonyukim** [comment]
Yeah!!! Thanks!
